jsonrpc/memconn.go

```diff
@@ -0,0 +1,29 @@
+package jsonrpc
+
+import "io"
+
+// MemConn is a simple in-memory JSON RPC connection.
+type MemConn struct {
```

Comment:
I think we can avoid adding this, we can keep HandleReader public and use that in tests.

Reply:
`Handle` is simple enough that we can skip testing it.

Reply:
> I think we can avoid adding this

Brilliant. Didn't think long enough to realize it wasn't necessary.

> we can keep HandleReader public and use that in tests.

Ended up using `net.Pipe()`, which is what the `MemConn` was mimicking anyway.